tensorboard/examples/plugins/example_raw_scalars/test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for the example plugin."""
+
+import os
+import ntpath
+import posixpath
+import unittest
+from unittest import mock
+
+from werkzeug import test
+from werkzeug import wrappers
+
+from tensorboard.plugins import base_plugin
+from tensorboard_plugin_example_raw_scalars import plugin
+
+
+def is_path_safe(path):
+    """Returns the result depending on the plugin's static file handler."""
+    example_plugin = plugin.ExampleRawScalarsPlugin(base_plugin.TBContext())
+    serve_static_file = example_plugin._serve_static_file
+
+    client = test.Client(serve_static_file, wrappers.BaseResponse)
+    response = client.get(plugin._PLUGIN_DIRECTORY_PATH_PART + path)
+    return response.status_code == 200
+
+
+class UrlSafetyTest(unittest.TestCase):
+    def test_path_traversal(self):
+        """Properly check whether a URL can be served from the static folder."""
+        with mock.patch(
+            "builtins.open", mock.mock_open(read_data="data")
+        ) as mock_file:
```

Comment:
You never use `mock_file`, so you don’t need the `as` clause.

Reply:
Done